docs/async/async-in-depth.md

```diff
@@ -0,0 +1,121 @@
+# Async In Depth
+
+By [Phillip Carter](https://github.com/cartermp)
+
+Writing I/O-Bound or CPU-Bound asynchronous code is simple using the `async` and `await` keywords, but what's happening under the covers is fairly complex.  The two key objects involved are `Task` and `Task<T>`.
+
+## Task and Task<T>
+
+Tasks are constructs used to represent operations running in the background.
```

Comment:
I believe that "in the background" is too specific. This doc doesn't use that terminology: https://msdn.microsoft.com/en-us/library/hh873175(v=vs.110).aspx